_rules/heading-descriptive-b49b2e.md

```diff
@@ -27,21 +27,11 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any [semantic][semantic role] `heading` element that is either [visible][] or [included in the accessibility tree][].
+This rule applies to any [semantic][semantic role] `heading` element that is [included in the accessibility tree][].
 
-## Expectation 1
+## Expectation
 
-For each target element at least one of the following is true:
-
-- the target element is not [visible][]; or
-- the target element describes the topic or purpose of the first [palpable content][] which is non-[decorative][], [visible][], and after the target element in tree order in the [flat tree][].
-
-## Expectation 2
-
-For each target element at least one of the following is true:
-
-- the target element is not [included in the accessibility tree][]; or
-- the target element describes the topic or purpose of the first [palpable content][] which is non-[decorative][], [included in the accessibility tree][], and after the target element in tree order in the [flat tree][].
```

Comment:
Good question.
I think [Palpable content](https://html.spec.whatwg.org/multipage/dom.html#palpable-content) tries to circumvent that with these bits:
> As a general rule, elements whose content model allows any [flow content](https://html.spec.whatwg.org/multipage/dom.html#flow-content-2) or [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2) should have at least one node in its [contents](https://html.spec.whatwg.org/multipage/dom.html#concept-html-contents) that is palpable content and that does not have the [hidden](https://html.spec.whatwg.org/multipage/interaction.html#attr-hidden) attribute specified.

> Note: [Palpable content](https://html.spec.whatwg.org/multipage/dom.html#palpable-content-2) makes an element non-empty by providing either some descendant non-empty [text](https://html.spec.whatwg.org/multipage/dom.html#text-content), or else something users can hear ([audio](https://html.spec.whatwg.org/multipage/media.html#the-audio-element) elements) or view ([video](https://html.spec.whatwg.org/multipage/media.html#the-video-element), [img](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element), or [canvas](https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element) elements) or otherwise interact with (for example, interactive form controls).

> This requirement is not a hard requirement, however, as there are many cases where an element can be empty legitimately, for example when it is used as a placeholder which will later be filled in by a script, or when the element is part of a template and would on most pages be filled in but on some pages is not relevant.

I read the fist bit as "elements listed here that allow flow content should be non-empty", which sort of cover your point (the embedded content elements such as `img` being likely assumed to also be non-empty).

But then the last paragraph states that this is not always true.

---
Maybe we can use our [perceivable content][https://act-rules.github.io/glossary/#perceivable-content) instead, which sort of tries to handle that?

Or go at length of "describe at least one node between this heading and the next of same or lower level", but that feels tricky…

Or we can try and use "visible palpable content", but adding a border to your `p` would also break that…

Reply:
@WilcoFiers worked on this today and the changes have been made.